changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -144,7 +145,210 @@ is to make the Hierarchy Keyring
 "Easy to Use" in a multiple KMS Relationship
 environment.
 
-## Relating Cache Entries (Branch Keys) to KMS Relationships
+## Summary
+
+Relevant sim [ticket](https://sim.amazon.com/issues/CrypTool-5311).
+
+To improve the MPL Consumer data key caching experience,
+when using the Hierarchy Keyring,
+we need to allow caching across KeyStores/KMS Clients/KMS Keys.
+
+To facilitate Caching across KeyStores/KMS Clients/KMS Keys,
+we MUST break the Cryptographic Materials Cache (CMC)
+out of the Hierarchy Keyring.
+
+By allowing MPL Consumers to provide an already initialized CMC
+to the Hierarchy Keyring at construction,
+the CMC MAY cache Branch Keys protected by different
+KMS Relationships.
+
+This simplifies Multiple KMS Relationship MPL Consumers,
+as they do not need to stand up LRU Cache of Hierarchy Keyrings.
+
+Instead, they may maintain one CMC.
+They still create a Hierarchy Keyring instance per KMS Relationship,
+and they MUST use the correct Keyring to retrieve material
+from the Cache.
+
+But they only need not maintain the common cache.
+
+Additionally, even though there is no CachingCMM in dafny right
+now, in addition to allowing a shared cache between multiple
+Hierarchy Keyrings, we want to create cache identifiers such
+that multiple CachingCMMs and multiple Hierarchy Keyrings can
+simultaneously use a single shared cache.
```

Comment:
Issue: This paragraph is hard to read.

Can you re-write it? 
I think the summary is:
> In the future, the CachingCMM will be introduced to Crypto Tool's Dafny products.
The CachingCMM and the Hierarchy Keyring both consume CMCs.
Thus, it will be possible to provide a CMC to both a Hierarchy Keyring & a CachingCMM.
This design MUST consider how the cache entries of multiple Hierarchy Keyrings and CachingCMMs will be appropriately separated.